### PR TITLE
PP-979: Upgrade from PBS Schema version 1.2 to 1.3 is failing

### PR DESCRIPTION
**Pull request by nithinj**
#### Issue-ID

* **[PP-979](https://pbspro.atlassian.net/browse/PP-979)**

#### Problem description
* Upgrade from PBS Schema version 1.2 to 1.3 is failing

#### Cause / Analysis
* Some of the checks in pbs_habitat was preventing the execution of the related function

#### Solution description
* Fixed issues in pbs_habitat

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

CC: @suresh-thelkar , @vinodchitrali , @bremanandjk 

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/cmds/scripts/pbs_habitat.in [hunk 1]
```diff
@@ -512,17 +512,7 @@ upgrade_pbs_schema() {
...
-		ret=$?
-		if [ $ret -ne 0 ]; then
-			return $ret
-		fi
-		return 1
-	fi
-
-
-	if [ "$ver" = "1.2.0" ]; then
+	elif [ "$ver" = "1.2.0" ]; then
```
Comment (vinodchitrali) [inline src/cmds/scripts/pbs_habitat.in]: change this to if

**Comment by nithinj** [comment]
Thanks @vinodchitrali and @suresh-thelkar for your review. I've addressed your comments in the recent commit. Please check.

**Comment by nithinj** [comment]
Thanks @suresh-thelkar and @ashwathraop for your sign-off.
I've attached logs to the ticket as @suresh-thelkar suggested.

@PBSPro/pbspro-maintainers ,
Please take this PR further.

File: src/cmds/scripts/pbs_habitat.in [hunk 2]
```diff
@@ -512,22 +512,16 @@ upgrade_pbs_schema() {
...
 			return $ret
 		fi
-	else
-		echo "Cannot upgrade PBS datastore version $ver"
-		ret=$?
-		if [ $ret -ne 0 ]; then
-			return $ret
-		fi
-		return 1
+		ver="1.2.0"
```
Comment (arungrover) [inline src/cmds/scripts/pbs_habitat.in]: In this block if we are taking care or both - v1.0.0 or v1.1.0 then can we remove the block of code from line 481 to 491 above which only works on 1.0.0? This seem redundant to me.

**Comment by nithinj** [comment]
Thanks @arungrover  for the review. I've squashed and re-based my branch. Please take this further.

**Comment by arungrover** [comment]
@nithinj you need another sign-off from reviewers who signed off earlier. I was hoping you would not have squashed it because now it will be difficult for reviewers to see the delta.

**Comment by arungrover** [comment]
@nithinj some other PRs made it in, please rebase it again.

**Comment by suresh-thelkar** [comment]
Code changes look good to me. I sign off.

**Comment by nithinj** [comment]
Hi @arungrover ,
I've re-based it. Please check.

**Comment by arungrover** [comment]
merged

**Pull request merged**